pkg/api/testing/unstructured_test.go

```diff
@@ -134,6 +142,208 @@ func TestRoundTrip(t *testing.T) {
+				internalObj, err := legacyscheme.Scheme.New(schema.GroupVersion{Group: gvk.Group, Version: runtime.APIVersionInternal}.WithKind(gvk.Kind))
+				if err != nil {
+					t.Fatalf("couldn't create internal object %v: %v", gvk.Kind, err)
+				}
+				fuzzer.Fuzz(internalObj)
+
+				item, err := legacyscheme.Scheme.New(gvk)
+				if err != nil {
+					t.Fatalf("couldn't create external object %v: %v", gvk.Kind, err)
+				}
+				if err := legacyscheme.Scheme.Convert(internalObj, item, nil); err != nil {
+					t.Fatalf("conversion for %v failed: %v", gvk.Kind, err)
+				}
+
+				// Decoding into Unstructured requires that apiVersion and kind be
+				// serialized, so populate TypeMeta.
+				item.GetObjectKind().SetGroupVersionKind(gvk)
+
+				jsonSerializer := jsonserializer.NewSerializerWithOptions(jsonserializer.DefaultMetaFactory, legacyscheme.Scheme, legacyscheme.Scheme, jsonserializer.SerializerOptions{})
+				cborSerializer := cborserializer.NewSerializer(legacyscheme.Scheme, legacyscheme.Scheme)
+
+				// original->JSON->Unstructured
+				buf.Reset()
+				if err := jsonSerializer.Encode(item, &buf); err != nil {
+					t.Fatalf("error encoding native to json: %v", err)
+				}
+				var uJSON runtime.Object = &metaunstruct.Unstructured{}
+				uJSON, _, err = jsonSerializer.Decode(buf.Bytes(), &gvk, uJSON)
+				if err != nil {
+					t.Fatalf("error decoding json to unstructured: %v", err)
+				}
+
+				// original->CBOR->Unstructured
+				buf.Reset()
+				if err := cborSerializer.Encode(item, &buf); err != nil {
+					t.Fatalf("error encoding native to cbor: %v", err)
+				}
+				var uCBOR runtime.Object = &metaunstruct.Unstructured{}
+				uCBOR, _, err = cborSerializer.Decode(buf.Bytes(), &gvk, uCBOR)
+				if err != nil {
+					diag, _ := cbor.Diagnose(buf.Bytes())
+					t.Fatalf("error decoding cbor to unstructured: %v, diag: %s", err, diag)
+				}
+
+				// original->JSON->Unstructured == original->CBOR->Unstructured
+				if !apiequality.Semantic.DeepEqual(uJSON, uCBOR) {
+					t.Fatalf("unstructured via json differed from unstructured via cbor: %v", cmp.Diff(uJSON, uCBOR))
+				}
+
+				// original->JSON/CBOR->Unstructured == original->JSON/CBOR->Unstructured->JSON->Unstructured
+				buf.Reset()
+				if err := jsonSerializer.Encode(uJSON, &buf); err != nil {
+					t.Fatalf("error encoding unstructured to json: %v", err)
+				}
+				var uJSON2 runtime.Object = &metaunstruct.Unstructured{}
+				uJSON2, _, err = jsonSerializer.Decode(buf.Bytes(), &gvk, uJSON2)
+				if err != nil {
+					t.Fatalf("error decoding json to unstructured: %v", err)
+				}
+				if !apiequality.Semantic.DeepEqual(uJSON, uJSON2) {
+					t.Errorf("object changed during native-json-unstructured-json-unstructured roundtrip, diff: %s", cmp.Diff(uJSON, uJSON2))
+				}
+
+				// original->JSON/CBOR->Unstructured == original->JSON/CBOR->Unstructured->CBOR->Unstructured
+				buf.Reset()
+				if err := cborSerializer.Encode(item, &buf); err != nil {
```

Comment:
should `item` be `uJSON` or `uCBOR`?

Reply:
Great catch, thanks! Yes, it should be one of those.